src/createApp.ts

```diff
@@ -899,26 +914,21 @@ const createApp = compose({
 			// custom elements.
 			const factories = widgetFactories.get(this);
 			const instances = widgetInstances.get(this);
-			return new Promise((resolve) => {
-				// First see if a factory exists for the widget.
-				let factory: WidgetFactory;
-				try {
-					factory = factories.get(id);
-				} catch (missingFactory) {
-					try {
-						// Otherwise try and get an existing instance.
-						const instance = instances.get(id);
-						resolve(instance);
-						return; // Make sure to return!
-					} catch (_) {
-						// Don't confuse people by complaining about missing instances, rethrow the
-						// original error.
-						throw missingFactory;
-					}
-				}
-				// This is only reached when a factory exists. Call it and resolve with the result.
-				// If it throws that's fine, it'll reject the promise.
-				resolve(factory());
+
+			let missingFactory: any;
+			return Promise.resolve().then(() => {
+				const factory = factories.get(id);
+				return factory();
+			})
+			.catch((e) => {
+				missingFactory = e;
```

Comment:
indeed, I wasn't keen on it either. Couldn't think of anything else that would be valuable, to be fair the error `Could not find a value for identity...` is mostly accurate.

if you explicitly added an entry to the app definition for the `id` of the widget I think you'd want it to use that configuration rather than a factory associated to a type on the state? And I do think that checking for an existing instance should occur before the a widget is attempted to be instantiated based on the type.